File: framework/test2/floating/matrix/matrix.php
<?php
require_once 'Soda/DS/index.php';
require_once 'Soda/Unittest/index.php';

use Soda\DS;
use Soda\Unittest\TestWork;
use Soda\Unittest\Utils;

class Solution {

    /**
     * @param Double[][] $a
     * @param Double[][] $b
     * @return Double[][]
     */
    function matrixMultiply($a, $b) {
        $rows = count($a);
        $cols = count($b[0]);
        $res = array_fill(0, $rows, array_fill(0, $cols, 0));
        foreach (range(0, $rows-1) as $i) {
            foreach (range(0, $cols-1) as $j) {
                $c = 0;
                foreach (range(0, count($b)-1) as $k) {
                    $c += $a[$i][$k] * $b[$k][$j];
                }
                $res[$i][$j] = $c;
            }
        }
        return $res;
    }
}

$work = TestWork::create(new Solution(), 'matrixMultiply', $argv[0]);
// $work->validator = fn($e, $r) => <boolean result>
// $work->compareSerial = true;
echo $work->run(Utils::fromStdin());

File: framework/test2/floating/matrix/Matrix.kt
import soda.kotlin.unittest.*

class Solution {
    fun matrixMultiply(a: Array<DoubleArray>, b: Array<DoubleArray>): Array<DoubleArray> {
        val rows = a.size
        val cols = b[0].size
        val res = Array(rows) { DoubleArray(cols) }
        for (i in 0 until rows) {
            for (j in 0 until cols) {
                var c = 0.0
                for (k in b.indices) {
                    c += a[i][k] * b[k][j]
                }
                res[i][j] = c
            }
        }
        return res
    }
}

class Matrix : (String) -> String {
    override fun invoke(text: String): String {
        // * setup with your method
        val work = GenericTestWork.create(Solution()::matrixMultiply)

        // * by method that has not return value
        // val work = GenericTestWork.createVoid(Solution.METHOD_WITHOUT_RETURN)
        // * by class of data struct
        // val work = GenericTestWork.forStruct(STRUCT::class)

        // * setup validator
        // work.validator = (T, T) -> Boolean
        // work.compareSerial = true
        return work.run(text)
    }

}

fun main() {
    print(Matrix()(Utils.fromStdin()))
}
